### PR TITLE
feat: add oneflow demo

### PR DESCRIPTION
**Pull request by Elliezza**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Add in a usecase demo for OpenMLDB + OneFlow
Supply demo scripts and sample datasets
Supply documentation in Chinese and English

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: docs/en/use_case/JD_recommendation_en.md [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+
+> χ may include categorical fields (e.g., gender, location) and continuous fields (e.g., age). Each categorical field is represented as a vector of one-hot encoding, and each continuous field is represented as the value itself, or a vector of one-hot encoding after discretization.
+
+Change directory to demo directory and execute the following commands to process the data set.
+```bash
+cd $demodir/openmldb_process/
+sh process_JD_out_full.sh $demodir/out/1
+```
+The generated dataset will be placed at `$demodir/openmldb_process/out`. After generating parquet dataset, dataset information will also be printed. It contains the information about the number of samples and table size array, which is needed when training.
+>train samples = 4007924
```
Comment (vagetablechicken) [inline docs/en/use_case/JD_recommendation_en.md:232]: After the data amount reduced, this print result is wrong

**Review by vagetablechicken — CHANGES_REQUESTED** [review]
Make sure that small dataset can do deepfm training.

File: docs/zh/use_case/JD_recommendation.md [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+python3 -m oneflow.distributed.launch \
+--nproc_per_node $DEVICE_NUM_PER_NODE \
+--nnodes 1 \
+--node_rank 0 \
+--master_addr 127.0.0.1 \
+deepfm_train_eval_JD.py \
+--disable_fusedmlp \
+--data_dir $DATA_DIR \
+--persistent_path $PERSISTENT_PATH \
+--table_size_array "11,42,1105,200,11,1295,1,1,5,3,23,23,7,5042381,3127923,5042381,3649642,28350,105180,7,2,5042381,5,4,4,41,2,2,8,3456,4,5,5042381,10,60,5042381,843,17,1276,101,100" \
```
Comment (vagetablechicken) [inline docs/zh/use_case/JD_recommendation.md]: 标注下table_size_array，num_train_samples等需要修改，或者直接参数化，又或者配置文本化

File: docs/zh/use_case/JD_recommendation.md [hunk 3]
```diff
@@ -0,0 +1,550 @@
...
+└── embedding
+    ├── 1
+    │   └── model
+    └── config.pbtxt
+ ```   
+#### 3.3.2 确认`config.pbtxt`中的配置正确。
+#### 3.3.3 确认persistent路径（`$demodir/oneflow_process/persistent`）正确
+ 
+### 3.4 启动推理服务
+#### 3.4.1 启动OpenMLDB推理服务
```
Comment (vagetablechicken) [inline docs/zh/use_case/JD_recommendation.md:434]: predict server也包括了访问triton，不只是openmldb推理服务，在后面新建一个章节

File: demo/JD-recommendation/oneflow_process/train_deepfm.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+DEVICE_NUM_PER_NODE=1
+DATA_DIR=/home/gtest/demo/openmldb_process/out
```
Comment (vagetablechicken) [inline demo/JD-recommendation/oneflow_process/train_deepfm.sh]: '/home/gtest/demo' is the root path of oneflow demo, make it as an argument

File: demo/JD-recommendation/openmldb_execution_steps.txt [hunk 5]
```diff
@@ -0,0 +1,119 @@
+##in openmldb docker
```
Comment (vagetablechicken) [inline demo/JD-recommendation/openmldb_execution_steps.txt]: This file is needless. Check all sql in *.sql

File: demo/JD-recommendation/openmldb_process/cal_table_array_size.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import pandas as pd
+import sys
+
+train_data = pd.read_parquet("/home/gtest/work/openmldb/new_process/out/train/train.parquet")
```
Comment (vagetablechicken) [inline demo/JD-recommendation/openmldb_process/cal_table_array_size.py]: fix the path

File: demo/JD-recommendation/openmldb_process/process_JD_out_full.sh [hunk 7]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+if [ ! -d "data_processed" ]
```
Comment (vagetablechicken) [inline demo/JD-recommendation/openmldb_process/process_JD_out_full.sh]: just use `mkdir -p`

File: demo/JD-recommendation/openmldb_process/process_JD_out_full.sh [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+number="$( find "$1"/*.csv | wc -l )"
+echo "total $number files"
+
+split1=$(( 8*number/10 ))
+split2=$(( 9*number/10 ))
+
+n=0
+echo "$split1 $split2"
+for f in "$1"/*.csv
```
Comment (vagetablechicken) [inline demo/JD-recommendation/openmldb_process/process_JD_out_full.sh:14]: It's better to use one python script to split dataset, and convert them.

File: demo/JD-recommendation/data/JD_data/openmldb_sql.txt [hunk 9]
```diff
@@ -0,0 +1,92 @@
+select * from
```
Comment (vagetablechicken) [inline demo/JD-recommendation/data/JD_data/openmldb_sql.txt:1]: Is this file needed?

File: demo/JD-recommendation/serving/README.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
+##in openmldb docker
```
Comment (vagetablechicken) [inline demo/JD-recommendation/serving/README.md]: Needless, move content into root README.

File: demo/JD-recommendation/serving/openmldb_serving.txt [hunk 11]
```diff
@@ -0,0 +1,103 @@
+##in openmldb docker
+/work/openmldb/bin/openmldb --zk_cluster=127.0.0.1:2181 --zk_root_path=/openmldb --role=sql_client
```
Comment (vagetablechicken) [inline demo/JD-recommendation/serving/openmldb_serving.txt]: Needless,  we've write all sql in *.sql.

File: docs/zh/use_case/JD_recommendation.md [hunk 12]
```diff
@@ -0,0 +1,552 @@
...
+    bo_action_pair_id_ingestionTime_0s_14d_100 as (
+UNION (select `ingestionTime`, `pair_id`, `time`, `model_id`, `type`, `cate`, `br`, '' as reqId from `bo_action`) partition by `pair_id` order by `ingestionTime` rows_range between 14d preceding and 0s preceding MAXSIZE 100 INSTANCE_NOT_IN_WINDOW))
+as out3
+on out0.reqId_1 = out3.reqId_17
+INTO OUTFILE '/root/project/out/1';
+```
+此处仅一个命令，可以使用阻塞式`LOAD DATA`，直接运行sql脚本`sync_select_out.sql`:
+
+```
+/work/openmldb/bin/openmldb --zk_cluster=127.0.0.1:2181 --zk_root_path=/openmldb --role=sql_client < /root/project/sync_select_out.sql
```
Comment (vagetablechicken) [inline docs/zh/use_case/JD_recommendation.md:211]: sync_select_out.sql != previous sql, it has `sync_job=true`, notice that

File: docs/zh/use_case/JD_recommendation.md [hunk 13]
```diff
@@ -0,0 +1,552 @@
...
+集群版的部分命令是非阻塞任务，包括在线模式的  `LOAD  DATA`，以及离线模式的  `LOAD  DATA`  ，`SELECT`，`SELECT  INTO`  命令。提交任务以后可以使用相关的命令如  `SHOW  JOBS`,  `SHOW  JOB`  来查看任务进度，详情参见离线任务管理文档。
+
+
+## 2. 机器学习训练流程
+### 2.1 流程概览
+使用OpenMLDB+OneFlow进行机器学习训练可总结为以下大致步骤。
+接下来会介绍每一个步骤的具体操作细节。
+
+### 2.2 使用OpenMLDB进行离线特征抽取
+#### 2.2.1 创建数据库和数据表
```
Comment (vagetablechicken) [inline docs/zh/use_case/JD_recommendation.md:71]: 说明为什么源数据是6张表，经由AutoML工具产生的，4张基本是原表，2张是为了数据标签而生产的表

Comment (vagetablechicken) [reply docs/zh/use_case/JD_recommendation.md:71]: 全量数据download链接https://openmldb.ai/download/jd-recommendation/JD_data.tgz

File: demo/JD-recommendation/sync_select_out.sql [hunk 14]
```diff
@@ -0,0 +1,88 @@
+USE JD_db;
+set @@sync_job='true';
+set @@job_timeout='600000';
```
Comment (Mario-LuX) [inline demo/JD-recommendation/sync_select_out.sql:3]: here timeout=600000ms, may meet with timeout error when exporting the whole JD dataset, but should be fine with small data

File: demo/JD-recommendation/load_data.sql [hunk 15]
```diff
@@ -0,0 +1,8 @@
+USE JD_db;
+SET @@execute_mode='offline';
```
Comment (Mario-LuX) [inline demo/JD-recommendation/load_data.sql:3]: Add SET @@job_timeout=600000;
The default timeout for every job is 20000ms(20 seconds), may meet with timeout error

File: demo/JD-recommendation/load_online_data.sql [hunk 16]
```diff
@@ -0,0 +1,8 @@
+USE JD_db;
+SET @@execute_mode='online';
```
Comment (Mario-LuX) [inline demo/JD-recommendation/load_online_data.sql:3]: Add SET @@job_timeout=600000;
The default timeout for every job is 20000ms(20 seconds), may meet with timeout error

File: demo/JD-recommendation/oneflow_process/train_deepfm.sh [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+    --node_rank 0 \
+    --master_addr 127.0.0.1 \
+    deepfm_train_eval_JD.py \
+      --disable_fusedmlp \
+      --data_dir "$DATA_DIR" \
+      --persistent_path "$PERSISTENT_PATH" \
+      --table_size_array "11,42,1105,200,11,1295,1,1,5,3,23,23,7,5042381,3127923,5042381,3649642,28350,105180,7,2,5042381,5,4,4,41,2,2,8,3456,4,5,5042381,10,60,5042381,843,17,1276,101,100" \
+      --store_type 'cached_host_mem' \
+      --cache_memory_budget_mb 1024 \
+      --batch_size 10000 \
```
Comment (Mario-LuX) [inline demo/JD-recommendation/oneflow_process/train_deepfm.sh:21]: After we change a smaller dataset, the --batch_size parameter should be changed to 1000 to make the test AUC not so bad

**Comment by Elliezza** [comment]
> Make sure that small dataset can do deepfm training.

Done.

**Review by vagetablechicken — APPROVED** [review]
we can merge it now, refactor later

File: demo/JD-recommendation/README.md [hunk 18]
```diff
@@ -0,0 +1,62 @@
+For full instructions, please refer to https://github.com/4paradigm/OpenMLDB/blob/main/docs/zh/use_case/JD_recommendation.md
```
Comment (dl239) [inline demo/JD-recommendation/README.md:1]: 404

File: demo/JD-recommendation/README.md [hunk 19]
```diff
@@ -0,0 +1,62 @@
+For full instructions, please refer to https://github.com/4paradigm/OpenMLDB/blob/main/docs/zh/use_case/JD_recommendation.md 
+
+#Training:
+1. Engage openmldb for feature extraction:
+##in openmldb docker
+docker exec -it demo bash
```
Comment (dl239) [inline demo/JD-recommendation/README.md:6]: how to download the `demo` image

**Review by dl239 — APPROVED** [review]

File: demo/JD-recommendation/README.md (hunk 19 above)
Comment (vagetablechicken) [reply demo/JD-recommendation/README.md:6]: readme is a bit old, I'll rewrite it in next pr.
